src/components/search/searchQueries.js

```diff
@@ -0,0 +1,141 @@
+/**
+ *
+ * @author sriram
+ *
+ *
+ */
+
+import { useQuery, useInfiniteQuery } from "react-query";
+import { searchData, searchDataInfinite } from "serviceFacades/filesystem";
+import { searchApps, searchAppsInfiniteQuery } from "serviceFacades/apps";
+import { getAnalyses, searchAnalysesInfinite } from "serviceFacades/analyses";
+
+/**
+ *
+ * Data search query
+ *
+ * @param {object} dataSearchKey - The query key to be used.
+ * @param {boolean} dataSearchQueryEnabled - Enable / diable query.
+ * @param {function} successCallback - Callback function to be used when the query succeeds.
+ *
+ * @returns {function}
+ */
+function useDataSearch(dataSearchKey, dataSearchQueryEnabled, successCallback) {
+    return useQuery({
+        queryKey: dataSearchKey,
+        queryFn: searchData,
+        config: {
+            enabled: dataSearchQueryEnabled,
+            onSuccess: (results) => successCallback(results),
```

Comment:
This `onSuccess` (and the others in this file) can be set directly to `successCallback`:
```suggestion
            onSuccess: successCallback,
```